Makefile

```diff
@@ -1,52 +1,191 @@
+	@$(INFO) Performing Docker build...
+	@docker build -f ${CONFIG_DOCKERFILE} . \
+	-t ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker build ${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-push
+docker-push: ## to push the docker image
+	@$(INFO) Pushing to registry...
+	@docker tag ${APP_NAME}:${APP_VERSION} $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@docker push $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Pushing to registry $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-sbom
+docker-sbom: ## to print a sbom report
+	@$(INFO) Performing Docker sbom report...
+	@docker sbom ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker sbom report
+
+.PHONY: docker-scan
+docker-scan: ## to print a vulnerability report
+	@$(INFO) Performing Docker scan report...
+	@docker scan ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker scan report
+
+.PHONY: docker-lint
+docker-lint: ## to lint the Dockerfile
+	@$(INFO) Dockerfile linting...
+	@docker run ${CONFIG_DOCKEROPTS} \
+	${IMAGE_DOCKERLINT} \
+	< ${CONFIG_DOCKERFILE} || ${FAIL}
+	@$(OK) Dockerfile linting
+
+go-build: $(GO_BUILD_PLATFORMS_ARTIFACTS) ## to build binaries
+
+.PHONY: go-build
+go-build/%:
+	@$(INFO) go build $*...
+	@target="$*"; \
+	command="$${target%%-*}"; \
+	platform_ext="$${target#*-}"; \
+	platform="$${platform_ext%.*}"; \
+	export GOOS="$${platform%%-*}"; \
+	export GOARCH="$${platform#*-}"; \
+	echo export GOOS=$${GOOS}; \
+	echo export GOARCH=$${GOARCH}; \
+	CGO_ENABLED=0 \
+	$(GO) build ${GO_BUILD_OPTS} \
+	-ldflags '${GO_LDFLAGS}' \
+	-o ${CONFIG_BIN_DIR}/$* \
+	${CONFIG_APP_CODE} || ${FAIL}
+	@$(OK) go build $*
+
+.PHONY: go-run
+go-run: ## to run locally for development
+	@$(INFO) running locally...
+	@$(GO) run ${GO_BUILD_OPTS} ${CONFIG_APP_CODE} || ${FAIL}
+	@$(OK) running locally
+
+.PHONY: go-test
+go-test: ## to run tests
+	@$(INFO) testing...
+	@$(GO) test ${GO_TEST_OPTS} ./... || ${FAIL}
+	@$(OK) testing
+
+.PHONY: go-mod-check
+go-mod-check: ## to check go mod files consistency
+	@$(INFO) Checking go mod files consistency...
+	@go mod tidy
+	@git --no-pager diff --exit-code go.mod go.sum || \
+	(${WARN} Please run "go mod tidy" and commit the changes in go.mod and go.sum. && ${FAIL} ; exit 128 )
+	@$(OK) Checking go mod files consistency
 
-	@[ "${tag}" ] || ( echo "docker-build requires a tag. Set a tag with: \"make docker-build tag=rtcd:my_tag\""; exit 128 )
+.PHONY: go-update-dependencies
+go-update-dependencies: ## to update go dependencies (vendor)
+	@$(INFO) updating go dependencies...
+	@$(GO) get -u ./... && \
+	$(GO) mod vendor && \
+	$(GO) mod tidy || ${FAIL}
+	@$(OK) updating go dependencies
 
-	docker build -f build/Dockerfile . -t $(tag)
+.PHONY: go-lint
+go-lint: ## to lint go code
+	@$(INFO) App linting...
+	@GOCACHE="/tmp" docker run ${CONFIG_DOCKEROPTS} \
```

Comment:
I didn't expect `go-lint` to run docker. Should we run the local binary instead? Either prompt to install it if missing or do it automatically.

Reply:
Hey @streamer45 Thanks for commenting. Really appreciate your input on this! 

There is a requirement to decouple the targets, from local tooling installed in the developer workstation 
Do check this comment more context: https://github.com/mattermost/rtcd/pull/17#discussion_r859538715